app/controllers/application_controller.rb

```diff
@@ -20,9 +20,57 @@ class ApplicationController < ActionController::Base
 
   def auth_user!
     unless user_signed_in? || devise_controller?
-      session[:previous_url] = request.fullpath unless request.xhr? # do not store AJAX calls
+      session[:previous_url] = request.fullpath unless request.xhr? # do not store AJAX call
       redirect_to new_user_session_path
     end
+    check_path
```

Comment:
Toto predsa nesuvisi s autorizaciou, nebolo by lepsie spravit vlastny before_filter?
A teda myslienka je, ze URL zostane rovnaka (nerobi sa redirect, iba sa prepisu params)? Najma to s tym prepisovanim params je celkom divoke a skor by sa o to mal starat controller, ktoreho sa to tyka (kam ta URL s miestnostami alebo tyzdnami smeruje), nez application controller globalne. Alebo je na to nejaky dovod?